e2e/test/provisioning/ProvisioningE2ETests.cs

```diff
@@ -896,44 +896,34 @@ public static ProvisioningClientOptions CreateProvisioningClientOptionsFromName(
         private static async Task ConfirmRegisteredDeviceWorksAsync(
             DeviceRegistrationResult result,
             IAuthenticationMethod auth,
-            IotHubClientTransportSettings transportSettings,
-            bool sendReportedPropertiesUpdate)
+            IotHubClientTransportSettings transportSettings)
         {
             await using var iotClient = new IotHubDeviceClient(result.AssignedHub, auth, new IotHubClientOptions(transportSettings));
             VerboseTestLogger.WriteLine("DeviceClient OpenAsync.");
-            await iotClient.OpenAsync().ConfigureAwait(false);
-            VerboseTestLogger.WriteLine("DeviceClient SendTelemetryAsync.");
-
-            var testMessage = new TelemetryMessage("TestMessage");
-            await iotClient.SendTelemetryAsync(testMessage).ConfigureAwait(false);
-
-            if (sendReportedPropertiesUpdate)
+            int attempt = 1;
+            while (true)
             {
-                VerboseTestLogger.WriteLine("DeviceClient updating desired properties.");
-                TwinProperties twin = await iotClient.GetTwinPropertiesAsync().ConfigureAwait(false);
-                var propertiesToReport = new ReportedProperties
+                try
+                {
+                    await iotClient.OpenAsync().ConfigureAwait(false);
+                    break;
+                }
+                catch (Exception ex) when (attempt < 4)
                 {
-                    [new Guid().ToString()] = new Guid().ToString(),
-                };
-                await iotClient.UpdateReportedPropertiesAsync(propertiesToReport).ConfigureAwait(false);
+                    VerboseTestLogger.WriteLine($"Attempt #{attempt} failed to open device {result.DeviceId} connection due to {ex}");
```

Comment:
Try a few times to open the client. We keep seeing an error in some PR builds where the device creds are rejected (so OpenAsync doesn't retry).